Userland/Libraries/LibCore/SharedCircularQueue.h

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Assertions.h>
+#include <AK/Atomic.h>
+#include <AK/BuiltinWrappers.h>
+#include <AK/Error.h>
+#include <AK/Format.h>
+#include <AK/Function.h>
+#include <AK/NonnullRefPtr.h>
+#include <AK/Platform.h>
+#include <AK/RefCounted.h>
+#include <AK/RefPtr.h>
+#include <AK/String.h>
+#include <AK/Types.h>
+#include <AK/Variant.h>
+#include <AK/Weakable.h>
+#include <LibCore/AnonymousBuffer.h>
+#include <LibCore/System.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <sched.h>
+#include <sys/mman.h>
+
+namespace Core {
+
+// A circular lock-free queue (or a buffer) with a single producer,
```

Comment:
This is something that really needs tests :smile:.

Reply:
abso-freaking-lutely.